gtk4/src/lib.rs

```diff
@@ -1,6 +1,161 @@
+//!             .build();
+//!
+//!         // Show the window.
+//!         win.show();
+//!     });
+//!
+//!     app.run();
+//! }
+//! ```
+//!
+//! # The main loop
+//!
+//! In a typical GTK application you set up the UI, assign signal handlers
+//! and run the main event loop.
+//!
+//! ```no_run
+
+//! use gtk4::prelude::*;
+//! use gtk4::{ApplicationBuilder, ApplicationWindowBuilder, Button};
+//!
+//! fn main() {
+//!     let application = ApplicationBuilder::new()
+//!         .application_id("com.example.FirstGtkApp")
+//!         .build()
+//!         .expect("Failed to create gtk application.");
+//!
+//!     application.connect_activate(|app| {
+//!         let window = ApplicationWindowBuilder::new()
+//!             .application(app)
+//!             .title("First GTK Program")
+//!             .default_width(350)
+//!             .default_height(70)
+//!             .build();
+//!
+//!         let button = Button::with_label("Click me!");
+//!         button.connect_clicked(|_| {
+//!             eprintln!("Clicked!");
+//!         });
+//!         window.set_child(Some(&button));
+//!
+//!         window.show();
+//!     });
+//!
+//!     application.run();
+//! }
+//! ```
+//!
+//! # Threads
+//!
+//! GTK is not thread-safe. Accordingly, none of this crate's structs implement
+//! [`Send`] or [`Sync`].
+//!
+//! The thread where [`fn@init`] was called is considered the main thread. OS X has
+//! its own notion of the main thread and [`fn@init`] must be called on that thread.
+//! After successful initialization, calling any [`gtk`](mod@crate) or [`mod@gdk`] functions
+//! (including [`fn@init`]) from other threads will `panic`.
+//!
+//! Any thread can schedule a closure to be run by the main loop on the main
+//! thread via [`fn@glib::idle_add`] or [`fn@glib::timeout_add`]. While
+//! working with GTK you might need the [`fn@glib::idle_add_local`]
+//! or [`fn@glib::timeout_add_local`] version without the
+//! [`Send`] bound. Those may only be called from the main thread.
+//!
+//! # Panics
+//!
+//! The [`gtk`](mod@crate) and [`mod@gdk`] crates have some run-time safety and contract checks.
+//!
+//! - Any constructor or free function will panic if called before [`fn@init`] or on
+//! a non-main thread.
+//!
+//! - Any [`&str`] or [`&Path`](std::path::Path) parameter with an interior null (`\0`) character will
+//! cause a panic.
+//!
+//! - Some functions will panic if supplied out-of-range integer parameters. All
+//! such cases will be documented individually but they are not yet.
+//!
+//! - A panic in a closure that handles signals or in any other closure passed
+//! to a [`gtk`](mod@crate) function will abort the process.
+//!
+//! # Features
+//!
+//! ## Library versions
+//!
+//! By default this crate provides only GTK 4.0 APIs. You can access additional
+//! functionality by selecting one of the `v4_2`, `v4_4`, etc. features.
+//!
+//! `Cargo.toml` example:
+//!
+//! ```toml
+//! [dependencies.gtk4]
```

Comment:
I tend to prefer renaming the package to gtk here and use that instead of doing `gtk4::`, especially that gsk/gdk are re-exported as  `gdk` `gsk` and not `gdk4`, `gsk4`. Maybe do that here as well?

Reply:
I agree. Changed it.

What do you think about the `gtk4` in the code examples above?

Reply:
should probably be renamed to use gtk as well, like everywhere else

Reply:
Renamed and tried to explain it